dll/win32/shell32/CShellLink.cpp

```diff
@@ -2832,6 +2838,51 @@ BOOL CShellLink::OnInitDialog(HWND hwndDlg, HWND hwndFocus, LPARAM lParam)
             SendMessageW(hRunCombo, CB_SETCURSEL, index, 0);
     }
 
+    BOOL disablecontrols = FALSE;
+    if (darwin)
+    {
+        disablecontrols = TRUE;
+        EnableWindow(GetDlgItem(hwndDlg, IDC_SHORTCUT_FIND), FALSE);
+        EnableWindow(GetDlgItem(hwndDlg, IDC_SHORTCUT_CHANGE_ICON), FALSE);
+    }
+    else
+    {
+        WCHAR path[MAX_PATH * 2];
+        path[0] = UNICODE_NULL;
+        HRESULT hr = GetPath(path, _countof(path), NULL, SLGP_RAWPATH);
+        if (FAILED(hr))
+            hr = GetPath(path, _countof(path), NULL, 0);
+#if DBG
+        if (GetKeyState(VK_CONTROL) < 0)
+            hr = path[0] = S_OK; // Allow inspection of PIDL parsing path
```

Comment:
I don't understand why this would need to assign to `path[0]` ?

Reply:
hr = S_OK pretends we are happy with the path even though it's really a non-filesystem ::{GUID} path. Setting `path[0]` just avoids the ASSERT that checks for this bad path.

Reply:
I would prefer (even if this is purely for debug) to not abuse `S_OK` for this and assign that `path[0]` a value like `UNICODE_NULL` or `'\0'`